### PR TITLE
Add USB converter for AT&T 56K series keyboards

### PR DESCRIPTION
**Pull request by engicoder**
## Description
Adding a USB converter for the AT&T 56K series keyboards. 

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/converter/att56k_usb/att56k.c [hunk 1]
```diff
@@ -0,0 +1,18 @@
+#include <stdbool.h>
+
+#include "att56k_if.h"
+#include "att56k.h"
+#include <print.h>
+#include "debug.h"
+#include "util.h"
```
Comment (drashna) [inline keyboards/converter/att56k_usb/att56k.c]: Most of these could be moved to the keyboard's header file, if they're even needed here, at all. ```suggestion
#include "att56k.h"
```

File: keyboards/converter/att56k_usb/att56k.h [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+ */
+
+
+enum custom_keycodes {
+  CLCK_TOG = SAFE_RANGE,
+  CLCK_CYC,
+  NEW_SAFE_RANGE //use "NEW_SAFE_RANGE" for keymap specific codes
+};
+
+#define KEYMAP( \
```
Comment (drashna) [inline keyboards/converter/att56k_usb/att56k.h]:
```suggestion
#define LAYOUT( \
```

File: keyboards/converter/att56k_usb/att56k.h [hunk 3]
```diff
@@ -0,0 +1,51 @@
+#ifndef ATT56K_H
```
Comment (drashna) [inline keyboards/converter/att56k_usb/att56k.h]:
```suggestion
#pragma once
```

File: keyboards/converter/att56k_usb/att56k.h [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+#define ATT56K_H
```
Comment (drashna) [inline keyboards/converter/att56k_usb/att56k.h]:
```suggestion
```

File: keyboards/converter/att56k_usb/att56k.h [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+    { K40  , K41  , K42  , K43  , K44  , K45  , K46  , K47  , K48   , K49  , K4A  , K4B  , K4C  , K4D  , K4E  , K4F  , }, \
+    { K50  , KC_NO, K52  , K53  , K54  , K55  , K56  , K57  , K58   , K59  , K5A  , K5B  , K5C  , K5D  , K5E  , K5F  , }, \
+    { K60  , K61  , K62  , K63  , K64  , K65  , K66  , K67  , K68   , K69  , K6A  , K6B  , K6C  , K6D  , K6E  , K6F  , }, \
+    { K70  , K71  , KC_NO, K73  , K74  , K75  , K76  , K77  , K78   , K79  , K7A  , K7B  , K7C  , K7D  , K7E  , K7F  , }  \
+}
+
+void on_scan(void);
+
+
+#endif
```
Comment (drashna) [inline keyboards/converter/att56k_usb/att56k.h]:
```suggestion
```

File: keyboards/converter/att56k_usb/att56k_if.c [hunk 6]
```diff
@@ -0,0 +1,362 @@
...
+COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+*/
+
+#include <stdbool.h>
+#include "att56k_if.h"
+#include <avr/io.h>
+#include <avr/interrupt.h>
+#include "print.h"
```
Comment (drashna) [inline keyboards/converter/att56k_usb/att56k_if.c]: Most of these could probably be removed.

File: keyboards/converter/att56k_usb/att56k_if.h [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+*/
+
+#ifndef ATT56K_IF_H
```
Comment (drashna) [inline keyboards/converter/att56k_usb/att56k_if.h]:
```suggestion
#pragma once
```

File: keyboards/converter/att56k_usb/att56k_if.h [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+#define ATT56K_IF_H
```
Comment (drashna) [inline keyboards/converter/att56k_usb/att56k_if.h]:
```suggestion
```

File: keyboards/converter/att56k_usb/att56k_if.h [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+    ATT56K_CLICK_MODE_COUNT,
+} att56k_click_mode_e;
+
+void    att56k_init(void);
+bool    att56k_has_data(void);
+uint8_t att56k_recv(void);
+void    att56k_click_cycle(void);
+bool    att56k_device_detected(void);
+
+#endif  // ATT56K_IF_H
```
Comment (drashna) [inline keyboards/converter/att56k_usb/att56k_if.h]:
```suggestion
```

File: keyboards/converter/att56k_usb/config.h [hunk 10]
```diff
@@ -0,0 +1,53 @@
...
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#ifndef CONFIG_H
```
Comment (drashna) [inline keyboards/converter/att56k_usb/config.h]:
```suggestion
#pragma once
```

File: keyboards/converter/att56k_usb/config.h [hunk 11]
```diff
@@ -0,0 +1,53 @@
...
+#define CONFIG_H
```
Comment (drashna) [inline keyboards/converter/att56k_usb/config.h]:
```suggestion
```

File: keyboards/converter/att56k_usb/config.h [hunk 12]
```diff
@@ -0,0 +1,53 @@
...
+
+/* key combination for command */
+#define IS_COMMAND() ( \
+    keyboard_report->mods == (MOD_BIT(KC_LSHIFT) | MOD_BIT(KC_RSHIFT)) \
+)
+
+
+
+
+#endif
```
Comment (drashna) [inline keyboards/converter/att56k_usb/config.h]:
```suggestion
```

File: keyboards/converter/att56k_usb/config.h [hunk 13]
```diff
@@ -0,0 +1,53 @@
...
+
+/* Mechanical locking support. Use KC_LCAP, KC_LNUM or KC_LSCR instead in keymap */
+//#define LOCKING_SUPPORT_ENABLE
+/* Locking resynchronize hack */
+//#define LOCKING_RESYNC_ENABLE
+
+/* key combination for command */
+#define IS_COMMAND() ( \
+    keyboard_report->mods == (MOD_BIT(KC_LSHIFT) | MOD_BIT(KC_RSHIFT)) \
+)
```
Comment (drashna) [inline keyboards/converter/att56k_usb/config.h]: This is set by default and should be removed.

File: keyboards/converter/att56k_usb/keymaps/default/keymap.c [hunk 14]
```diff
@@ -0,0 +1,34 @@
+
+#include <stdint.h>
+#include <stdbool.h>
+#include "keycode.h"
+#include "print.h"
+#include "debug.h"
+#include "util.h"
+#include "att56k.h"
```
Comment (drashna) [inline keyboards/converter/att56k_usb/keymaps/default/keymap.c]: This should be replaced with: 
```suggestion
#include QMK_KEYBOARD_H
```

File: keyboards/converter/att56k_usb/keymaps/default/keymap.c [hunk 15]
```diff
@@ -0,0 +1,34 @@
...
+
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+	KEYMAP(
```
Comment (drashna) [inline keyboards/converter/att56k_usb/keymaps/default/keymap.c]:
```suggestion
	[0] = LAYOUT(
```

File: keyboards/converter/att56k_usb/keymaps/default/keymap.c [hunk 16]
```diff
@@ -0,0 +1,34 @@
+
+#include <stdint.h>
+#include <stdbool.h>
+#include "keycode.h"
+#include "print.h"
+#include "debug.h"
+#include "util.h"
+#include "att56k.h"
+
+
```
Comment (drashna) [inline keyboards/converter/att56k_usb/keymaps/default/keymap.c]: There should be a note up here to use `NEW_SAFE_RANG` for any macros.

File: keyboards/converter/att56k_usb/keymaps/default/keymap.c [hunk 17]
```diff
@@ -0,0 +1,34 @@
...
+    switch (keycode) {
+        case CLCK_CYC:
+            if (record->event.pressed) {
+                att56k_click_cycle();
+            }
+            return false;
+            break;
+    }
+    return true;
+}
```
Comment (drashna) [inline keyboards/converter/att56k_usb/keymaps/default/keymap.c]: This seems like it should be in the keyboard's c file, as `process_record_kb` instead.

File: keyboards/converter/att56k_usb/rules.mk [hunk 18]
```diff
@@ -0,0 +1,78 @@
...
+# Boot Section Size in *bytes*
+#   Atmel DFU loader(ATmega32U4)   4096
+#   Atmel DFU loader(AT90USB128)   8192
+#   LUFA bootloader(ATmega32U4)    4096
+#   Arduino Caterina(ATmega32U4)   4096
+#   USBaspLoader(ATmega***)        2048
+#   Teensy   halfKay(ATmega32U4)   512
+#   Teensy++ halfKay(AT90USB128)   2048
+OPT_DEFS += -DBOOTLOADER_SIZE=4096
+#BOOTLOADER = catarina
```
Comment (drashna) [inline keyboards/converter/att56k_usb/rules.mk]: if this is a pro micro, as it seems to be, this entire section should be replaced with just: 
```makefile
BOOTLOADER = catarina
```

**Comment by engicoder** [comment]
Committed changes requested. The files att56k.h and att56k.c had to be renamed to att56k_usb.h and att56k_usb.c to work properly with the build and the QMK_KEYBOARD_H macro.

File: keyboards/converter/att56k_usb/rules.mk [hunk 19]
```diff
@@ -0,0 +1,41 @@
...
+
+# Bootloader selection
+#   Teensy       halfkay
+#   Pro Micro    caterina
+#   Atmel DFU    atmel-dfu
+#   LUFA DFU     lufa-dfu
+#   QMK DFU      qmk-dfu
+#   ATmega32A    bootloadHID
+#   ATmega328P   USBasp
+BOOTLOADER = catarina
```
Comment (fauxpark) [inline keyboards/converter/att56k_usb/rules.mk]:
```suggestion
BOOTLOADER = caterina
```

File: keyboards/converter/att56k_usb/rules.mk [hunk 20]
```diff
@@ -0,0 +1,41 @@
...
+
+# Build Options
+#   change to "no" to disable the options, or define them in the Makefile in
+#   the appropriate keymap folder that will get included automatically
+#
+BOOTMAGIC_ENABLE = no       # Virtual DIP switch configuration(+1000)
```
Comment (fauxpark) [inline keyboards/converter/att56k_usb/rules.mk]:
```suggestion
BOOTMAGIC_ENABLE = no       # Virtual DIP switch configuration
```

File: keyboards/converter/att56k_usb/rules.mk [hunk 21]
```diff
@@ -0,0 +1,41 @@
...
+MOUSEKEY_ENABLE = yes       # Mouse keys(+4700)
```
Comment (fauxpark) [inline keyboards/converter/att56k_usb/rules.mk]:
```suggestion
MOUSEKEY_ENABLE = yes       # Mouse keys
```

File: keyboards/converter/att56k_usb/rules.mk [hunk 22]
```diff
@@ -0,0 +1,41 @@
...
+EXTRAKEY_ENABLE = yes        # Audio control and System control(+450)
```
Comment (fauxpark) [inline keyboards/converter/att56k_usb/rules.mk]:
```suggestion
EXTRAKEY_ENABLE = yes        # Audio control and System control
```

File: keyboards/converter/att56k_usb/rules.mk [hunk 23]
```diff
@@ -0,0 +1,41 @@
...
+CONSOLE_ENABLE = yes        # Console for debug(+400)
```
Comment (fauxpark) [inline keyboards/converter/att56k_usb/rules.mk]:
```suggestion
CONSOLE_ENABLE = yes        # Console for debug
```

File: keyboards/converter/att56k_usb/README.md [hunk 24]
```diff
@@ -0,0 +1,75 @@
...
+DATA     \__/                               \___/       \___/   \___/
+     _____________________________________________________________________          __
+Click                                                                     \________/
+```
+DATA transitions low half a clock cycle before CLOCK to ensure the leading 0 bit value. 
+## Build
+
+From the root qmk_firmware directory execute:
+
+    $ make converter/att56k_usb
```
Comment (fauxpark) [inline keyboards/converter/att56k_usb/README.md]:
```suggestion
    make converter/att56k_usb:default
```

File: keyboards/converter/att56k_usb/README.md [hunk 25]
```diff
@@ -0,0 +1,75 @@
...
+CLOCK     \_/ \_/ \_/ \_/ \_/ \_/ \_/ \_/ \_/ \_/ \_/ \_/ \_/ \_/ \_/ \_/
+     ____    _______________________________     _______     ___     _________________
+DATA     \__/                               \___/       \___/   \___/
+     _____________________________________________________________________          __
+Click                                                                     \________/
+```
+DATA transitions low half a clock cycle before CLOCK to ensure the leading 0 bit value. 
+## Build
+
+From the root qmk_firmware directory execute:
```
Comment (fauxpark) [inline keyboards/converter/att56k_usb/README.md]:
```suggestion
Make example for this keyboard (after setting up your build environment):
```

File: keyboards/converter/att56k_usb/README.md [hunk 26]
```diff
@@ -0,0 +1,75 @@
...
+
+    $ make converter/att56k_usb
+
+To specify a custom keymap in the folder <custom_keymap>, execute
```
Comment (fauxpark) [inline keyboards/converter/att56k_usb/README.md]:
```suggestion
```

File: keyboards/converter/att56k_usb/README.md [hunk 27]
```diff
@@ -0,0 +1,75 @@
...
+
+    $ make converter/att56k_usb:custom_keymap
```
Comment (fauxpark) [inline keyboards/converter/att56k_usb/README.md]:
```suggestion
```

File: keyboards/converter/att56k_usb/README.md [hunk 28]
```diff
@@ -0,0 +1,75 @@
...
+```
+DATA transitions low half a clock cycle before CLOCK to ensure the leading 0 bit value. 
+## Build
+
+From the root qmk_firmware directory execute:
+
+    $ make converter/att56k_usb
+
+To specify a custom keymap in the folder <custom_keymap>, execute
+
```
Comment (fauxpark) [inline keyboards/converter/att56k_usb/README.md]:
```suggestion
```

File: keyboards/converter/att56k_usb/README.md [hunk 29]
```diff
@@ -0,0 +1,75 @@
...
+     _____________________________________________________________________          __
+Click                                                                     \________/
+```
+DATA transitions low half a clock cycle before CLOCK to ensure the leading 0 bit value. 
+## Build
+
+From the root qmk_firmware directory execute:
+
+    $ make converter/att56k_usb
+
```
Comment (fauxpark) [inline keyboards/converter/att56k_usb/README.md]:
```suggestion
```

**Comment by engicoder** [comment]
Commited changes requested by @fauxpark

**Comment by engicoder** [comment]
I made the requested changes a month ago. Is there something else I should be doing to get this reviewed?

File: keyboards/converter/att56k_usb/rules.mk [hunk 30]
```diff
@@ -0,0 +1,41 @@
...
+UNICODE_ENABLE = no         # Unicode
+UNICODEMAP_ENABLE = no      # Unicode keymap
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+RGBLIGHT_ENABLE = no        # Enable WS2812 RGB underlight.  Do not enable this with audio at the same time.
+API_SYSEX_ENABLE = no
+CUSTOM_MATRIX = yes
+DEBUG_ENABLE = yes
+
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no    # Breathing sleep LED during USB suspend
```
Comment (noroadsleft) [inline keyboards/converter/att56k_usb/rules.mk:36]: Aligns to QMK's AVR `rules.mk` template.

```suggestion
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = yes           # USB Nkey Rollover 
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
MIDI_ENABLE = no            # MIDI controls
AUDIO_ENABLE = no           # Audio output on port C6
UNICODE_ENABLE = no         # Unicode
UNICODEMAP_ENABLE = no      # Unicode keymap
BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
MIDI_ENABLE = no            # MIDI support
BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
AUDIO_ENABLE = no           # Audio output on port C6
FAUXCLICKY_ENABLE = no      # Use buzzer to emulate clicky switches
HD44780_ENABLE = no         # Enable support for HD44780 based LCDs
API_SYSEX_ENABLE = no
CUSTOM_MATRIX = yes
DEBUG_ENABLE = yes
```

File: keyboards/converter/att56k_usb/att56k_usb.h [hunk 31]
```diff
@@ -0,0 +1,48 @@
...
+  CLCK_CYC = SAFE_RANGE,
+  NEW_SAFE_RANGE //use "NEW_SAFE_RANGE" for keymap specific codes
+};
+
+#define LAYOUT( \
+                        K3C, K08, K56, K03, K24, K06, K04, K30, K7C, K3D, K49, K74, \
+                        K0E, K38, K1D, K5F, K79, K6C, K53, K7A, K1F, K3F, K2F, K2B, \
+  K1A, K2C,   K20, K58, K09, K3E, K7E, K10, K2E, K3A, K4C, K5E, K0F, K27, K2D,   K4B,     K57, K7B, K4E,   K12, K14, K07, K7F, \
+  K26, K37,     K46,  K3B, K23, K33, K76, K16, K34, K40, K6B, K48, K66, K75, K69,  K21,   K39, K42, K68,   K64, K0A, K5C, K4A, \
+  K43, K44,      K19,  K35, K29, K63, K70, K22, K28, K05, K65, K73, K60, K5A, K6F,             K2A,        K4D, K6D, K50, K0C, \
```
Comment (noroadsleft) [inline keyboards/converter/att56k_usb/att56k_usb.h:32]: QMK custom is that the Enter keycode should be on the home row.

```suggestion
  K26, K37,     K46,  K3B, K23, K33, K76, K16, K34, K40, K6B, K48, K66, K75, K69,         K39, K42, K68,   K64, K0A, K5C, K4A, \
  K43, K44,      K19,  K35, K29, K63, K70, K22, K28, K05, K65, K73, K60, K5A, K6F, K21,        K2A,        K4D, K6D, K50, K0C, \
```

File: keyboards/converter/att56k_usb/info.json [hunk 32]
```diff
@@ -0,0 +1,136 @@
...
+        {"label":"R", "x":6.75, "y":4},
+        {"label":"T", "x":7.75, "y":4},
+        {"label":"Y", "x":8.75, "y":4},
+        {"label":"U", "x":9.75, "y":4},
+        {"label":"I", "x":10.75, "y":4},
+        {"label":"O", "x":11.75, "y":4},
+        {"label":"P", "x":12.75, "y":4},
+        {"label":"[", "x":13.75, "y":4},
+        {"label":"\\", "x":14.75, "y":4},
+        {"label":"Return", "x":15.75, "y":4, "w":1.5, "ks":[ [0,0], [1.5,0], [1.5,2], [0.25,2], [0.25,1], [0,1], [0,0] ]},
```
Comment (noroadsleft) [inline keyboards/converter/att56k_usb/info.json:72]: Moving this...

```suggestion
```

File: keyboards/converter/att56k_usb/info.json [hunk 33]
```diff
@@ -0,0 +1,136 @@
...
+        {"label":"F", "x":7, "y":5},
+        {"label":"G", "x":8, "y":5},
+        {"label":"H", "x":9, "y":5},
+        {"label":"J", "x":10, "y":5},
+        {"label":"K", "x":11, "y":5},
+        {"label":"L", "x":12, "y":5},
+        {"label":";", "x":13, "y":5},
+        {"label":"'", "x":14, "y":5},
+        {"label":"{", "x":15, "y":5},
+        {"label":"Up", "x":18.5, "y":5},
```
Comment (noroadsleft) [inline keyboards/converter/att56k_usb/info.json:95]: Configurator's UI code also doesn't understand the `ks` key, so our custom is to render ISO Enters as 1.25x2.

```suggestion
        {"label":"Return", "x":16, "y":4, "w":1.25, "h":2},
        {"label":"Up", "x":18.5, "y":5},
```

File: keyboards/converter/att56k_usb/info.json [hunk 34]
```diff
@@ -0,0 +1,136 @@
...
+        {"label":"WS Ctrol", "x":1, "y":7},
+        {"label":"Rese", "x":2.25, "y":7, "w":1.5},
+        {"label":"Alt/Ctrk", "x":4.75, "y":7, "w":1.5},
+        {"label":"Space", "x":6.25, "y":7, "w":7},
+        {"label":"Alt/Ctrl", "x":13.25, "y":7, "w":1.5},
+        {"label":"Enter", "x":15.75, "y":7, "w":1.5},
+        {"label":"Down", "x":18.5, "y":7},
+        {"label":"No", "x":20.75, "y":7},
+        {"label":"0", "x":21.75, "y":7, "w":2},
+        {"label":".", "x":23.75, "y":7}
```
Comment (noroadsleft) [inline keyboards/converter/att56k_usb/info.json:132]: You have an extra key here.

```suggestion
        {"label":"0", "x":20.75, "y":7, "w":2},
        {"label":".", "x":22.75, "y":7}
```

File: keyboards/converter/att56k_usb/keymaps/default/keymap.c [hunk 35]
```diff
@@ -0,0 +1,24 @@
...
+ * Note: Use NEW_SAFE_RANGE to define new custom keycodes and macros as
+ * SAFE_RANGE was used to define CLCK_CYC */
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+	[0] = LAYOUT(
+                               KC_F13, KC_F14, KC_F15, KC_F16, KC_F17, KC_F18, KC_F19, KC_F20, KC_F21, KC_F22, KC_F23, CLCK_CYC,
+                               KC_F1,  KC_F2,  KC_F3,  KC_F4,  KC_F5,  KC_F6,  KC_F7,  KC_F8,  KC_F9,  KC_F10, KC_F11, KC_F12,
+  KC_F1  , KC_F2  ,   KC_GRV , KC_1   , KC_2   , KC_3   , KC_4   , KC_5   , KC_6   , KC_7   , KC_8   , KC_9   , KC_0   , KC_MINS, KC_EQL , KC_BSPC,   KC_INS , KC_HOME, KC_PGUP,    KC_ESC , KC_PSLS, KC_PCMM, KC_SPC,
+  KC_F3  , KC_F4  ,    KC_TAB , KC_Q   , KC_W   , KC_E   , KC_R   , KC_T   , KC_Y   , KC_U   , KC_I   , KC_O   , KC_P  , KC_LBRC, KC_RBRC, KC_ENT ,   KC_DEL , KC_END , KC_PGDN,    KC_P7  , KC_P8  , KC_P9  , KC_TAB,
+  KC_F5  , KC_F6  ,     KC_CAPS, KC_A   , KC_S   , KC_D   , KC_F   , KC_G   , KC_H   , KC_J  , KC_K   , KC_L   , KC_SCLN, KC_QUOT, KC_SLSH,                    KC_UP  ,             KC_P4  , KC_P5  , KC_P6  , KC_PMNS,
```
Comment (noroadsleft) [inline keyboards/converter/att56k_usb/keymaps/default/keymap.c:18]:
```suggestion
  KC_F3  , KC_F4  ,    KC_TAB , KC_Q   , KC_W   , KC_E   , KC_R   , KC_T   , KC_Y   , KC_U   , KC_I   , KC_O   , KC_P  , KC_LBRC, KC_RBRC,            KC_DEL , KC_END , KC_PGDN,    KC_P7  , KC_P8  , KC_P9  , KC_TAB,
  KC_F5  , KC_F6  ,     KC_CAPS, KC_A   , KC_S   , KC_D   , KC_F   , KC_G   , KC_H   , KC_J  , KC_K   , KC_L   , KC_SCLN, KC_QUOT, KC_SLSH, KC_ENT,            KC_UP  ,             KC_P4  , KC_P5  , KC_P6  , KC_PMNS,
```

File: keyboards/converter/att56k_usb/README.md [hunk 36]
```diff
@@ -0,0 +1,67 @@
...
+  - Converter: Pro Micro 5V, Teensy 2.0, ATmega32u4
+  - Keyboards: 
+    - AT&T 56K 460 ACW - 122 key magnetic separation
+    - AT&T 56K 560 AGA - 122 key buckling spring
+
+
+## Interfacing
+
+The 56k keyboards use a 16 bit data packet. Depending on the keyboard the 
+transmission maybe synchronous with a 9.6Kz clock frequency or ansynchronous
```
Comment (tzarc) [inline keyboards/converter/att56k_usb/README.md:15]:
```suggestion
transmission may be synchronous with a 9.6kHz clock frequency or asynchronous
```

File: keyboards/converter/att56k_usb/att56k_if.c [hunk 37]
```diff
@@ -0,0 +1,362 @@
+/*
```
Comment (tzarc) [inline keyboards/converter/att56k_usb/att56k_if.c:1]: This file uses a lot of AVR-specific register definitions and the like. I do understand that the interrupts are AVR-specific, and at this stage we don't have a good abstraction over the top of these.
However, in general we'd prefer to see the GPIO operations converted over to use the platform-agnostic APIs, such as `setPinOutput()` and `writePin()` instead.

File: keyboards/converter/att56k_usb/att56k_usb.c [hunk 38]
```diff
@@ -0,0 +1,13 @@
+
```
Comment (tzarc) [inline keyboards/converter/att56k_usb/att56k_usb.c:1]: Needs a license header.

File: keyboards/converter/att56k_usb/att56k_usb.h [hunk 39]
```diff
@@ -0,0 +1,47 @@
+#pragma once
```
Comment (tzarc) [inline keyboards/converter/att56k_usb/att56k_usb.h:1]: Needs a license header.

File: keyboards/converter/att56k_usb/config.h [hunk 40]
```diff
@@ -0,0 +1,53 @@
...
+#define CONFIG_H
+
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0xFEED
+#define PRODUCT_ID      0x6060
+#define DEVICE_VER      0x0001
+#define MANUFACTURER    AT&T
+#define PRODUCT         56K 460 ACW
+#define DESCRIPTION     AT&T 56K 460 ACW Keyboard
```
Comment (tzarc) [inline keyboards/converter/att56k_usb/config.h:28]:
```suggestoin
```

File: keyboards/converter/att56k_usb/keymaps/default/keymap.c [hunk 41]
```diff
@@ -0,0 +1,23 @@
+
```
Comment (tzarc) [inline keyboards/converter/att56k_usb/keymaps/default/keymap.c:1]: Needs a license header.

File: keyboards/converter/att56k_usb/matrix.c [hunk 42]
```diff
@@ -0,0 +1,130 @@
...
+
+#define TX_LED_INIT() (DDRB |= (1 << PB0))
+#define TX_LED_ON() (PORTB &= ~(1 << PB0))
+#define TX_LED_OFF() (PORTB |= (1 << PB0))
+#define TX_LED_TOGGLE() (PORTB ^= (1 << PB0))
+
+#ifndef DEBOUNCE
+#    define DEBOUNCE 0
+#endif
+
```
Comment (tzarc) [inline keyboards/converter/att56k_usb/matrix.c:44]:
```suggestion
```

File: keyboards/converter/att56k_usb/rules.mk [hunk 43]
```diff
@@ -0,0 +1,45 @@
...
+MCU = atmega32u4
+
+# Bootloader selection
+#   Teensy       halfkay
+#   Pro Micro    caterina
+#   Atmel DFU    atmel-dfu
+#   LUFA DFU     lufa-dfu
+#   QMK DFU      qmk-dfu
+#   ATmega32A    bootloadHID
+#   ATmega328P   USBasp
```
Comment (tzarc) [inline keyboards/converter/att56k_usb/rules.mk:11]:
```suggestion
```

File: keyboards/converter/att56k_usb/rules.mk [hunk 44]
```diff
@@ -0,0 +1,45 @@
...
+MOUSEKEY_ENABLE = yes       # Mouse keys
+EXTRAKEY_ENABLE = yes        # Audio control and System control
+CONSOLE_ENABLE = yes        # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = yes           # USB Nkey Rollover 
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+MIDI_ENABLE = no            # MIDI controls
```
Comment (tzarc) [inline keyboards/converter/att56k_usb/rules.mk:27]:
```suggestion
```

File: keyboards/converter/att56k_usb/rules.mk [hunk 45]
```diff
@@ -0,0 +1,45 @@
...
+AUDIO_ENABLE = no           # Audio output on port C6
+UNICODE_ENABLE = no         # Unicode
+UNICODEMAP_ENABLE = no      # Unicode keymap
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+MIDI_ENABLE = no            # MIDI support
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+AUDIO_ENABLE = no           # Audio output on port C6
+FAUXCLICKY_ENABLE = no      # Use buzzer to emulate clicky switches
+HD44780_ENABLE = no         # Enable support for HD44780 based LCDs
```
Comment (tzarc) [inline keyboards/converter/att56k_usb/rules.mk:37]:
```suggestion
```

**Pull request closed without merging**